### ISSUE TITLE
About SHA256 OID in AlgorithmIdentifier

### ISSUE THREAD

**Issue by hug-dev** [Devolutions/picky-rs#19]
Hi again 😄 

We are trying to implement the following ASN.1 type:
```
DigestInfo ::= SEQUENCE {
    digestAlgorithm DigestAlgorithmIdentifier,
    digest Digest
}

DigestAlgorithmIdentifier ::= AlgorithmIdentifier
Digest ::= OCTET STRING
```
as the library we are using expect that structure to be sent as digest when verifying signatures. We would like `digestAlgorithm` to be a SHA256 OID (`2.16.840.1.101.3.4.2.1`) but it's not in the list of [constructors](https://docs.rs/picky/4.5.1/picky/algorithm_identifier/struct.AlgorithmIdentifier.html)!
Would it make sense to add it? I would be happy to make a PR 😃

**Comment by bcortier-devolutions** [comment]
Hi!

I believe it makes sense! :smiley: 
AlgorithmIdentifier might be one of the structure we'll want to port to a separate crate like we discussed previously in https://github.com/Devolutions/picky-rs/issues/12 and I think it can't hurt to support more.

You would need to add the OID definition after this line: https://github.com/Devolutions/picky-rs/blob/master/picky/src/oids.rs#L90 (I ordered by increasing OID)

Probably something like
```
// secure hash algorithms
SHA256 => sha256 => 2.16.840.1.101.3.4.2.1
...
```
I generally follow the [official naming]( https://oidref.com/2.16.840.1.101.3.4.2.1). It wouldn't hurt to also add brothers OID if you have time (this is mostly a repetitive cask haha).

Thank you :smile:
